app/src/main/java/com/kenchen/capstonenewsapp/views/news/NewsViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package com.kenchen.capstonenewsapp.views.news
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.kenchen.capstonenewsapp.App.Companion.remoteApi
+import com.kenchen.capstonenewsapp.model.Article
+import com.kenchen.capstonenewsapp.networking.RemoteResult
+import com.kenchen.capstonenewsapp.utils.mapException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class NewsViewModel : ViewModel() {
+
+    private val _headLineNewsLiveData = MutableLiveData<RemoteResult<List<Article>>>()
+    val headLineNewsLiveData: LiveData<RemoteResult<List<Article>>> = _headLineNewsLiveData
+
+    private val test = MutableLiveData<List<String>>()
+
+    private val _newsLoadingStateLiveData = MutableLiveData<NewsLoadingState>()
+    val newsLoadingStateLiveData: LiveData<NewsLoadingState> = _newsLoadingStateLiveData
+
+    fun onActivityReady() {
+        // prevent refetching data when orientation changes (save resource)
+        if (_headLineNewsLiveData.value == null) {
+            getTopHeadlinesNewsByCountry()
+        }
+    }
+
+    fun refreshNews() {
+        getTopHeadlinesNewsByCountry()
+    }
+
+    private fun getTopHeadlinesNewsByCountry() {
+        _newsLoadingStateLiveData.value = NewsLoadingState.LOADING
+        viewModelScope.launch(Dispatchers.IO) {
+            try {
+                val result = remoteApi.getTopHeadlinesByCountry("us")
+
+                // use postValue instead of value because
+                // we are on IO dispatcher and not on the
+                // Main (UI) thread
+                _headLineNewsLiveData.postValue(RemoteResult.Success(result))
+                _newsLoadingStateLiveData.postValue(NewsLoadingState.LOADED)
```

Comment:
Great work here! One alternative is to switch the coroutine context back to the `Main` dispatcher.  Then you could use `value` instead of `postValue()`:

```kotlin
withContext(Dispatchers.Main) {
    _headlineNewsLIveData.value = RemoteResult.Success(result)
    _newsLoadingStateLiveData.value = NewsLoadingState.LOADED
}
```

But there's nothing wrong with the approach you provided above either :]